File: CS/DataBindingCapabilitiesConverters/DataBindingViaDefaultConvertersFluentAPIUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSampleDataBindingCapabilitiesConverters {
    public partial class DataBindingViaDefaultConvertersFluentAPIUserControl : XtraUserControl {
        public DataBindingViaDefaultConvertersFluentAPIUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            TrackBarControl trackBar = new TrackBarControl();
            trackBar.Dock = DockStyle.Top;
            trackBar.Properties.Minimum = 0;
            trackBar.Properties.Maximum = 100;
        
            TextEdit editor = new TextEdit();
            editor.Dock = DockStyle.Top;
            editor.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric;
            editor.Properties.Mask.EditMask = "P0";
            editor.Properties.Mask.UseMaskAsDisplayFormat = true;
        
            editor.Parent = this;
            trackBar.Parent = this;
            #endregion SetUp

            #region #dataBindingViaDefaultConvertersFluentAPI
            // Set type of POCO-ViewModel
            mvvmContext.ViewModelType = typeof(ViewModel);
            // Data binding for the Progress property (via MVVMContext FluentAPI)
            var fluent = mvvmContext.OfType<ViewModel>();
            fluent.SetBinding(trackBar, e => e.EditValue, x => x.Progress);
            fluent.SetBinding(editor, e => e.EditValue, x => x.Progress);
            #endregion #dataBindingViaDefaultConvertersFluentAPI
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        public class ViewModel {
            public virtual int Progress { get; set; }
        }
    }
}

File: VB/DataBindingCapabilitiesConverters/DataBindingViaDefaultConvertersFluentAPIUserControl.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSampleDataBindingCapabilitiesConverters
    Partial Public Class DataBindingViaDefaultConvertersFluentAPIUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New()
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me

            Dim trackBar As New TrackBarControl()
            trackBar.Dock = DockStyle.Top
            trackBar.Properties.Minimum = 0
            trackBar.Properties.Maximum = 100

            Dim editor As New TextEdit()
            editor.Dock = DockStyle.Top
            editor.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric
            editor.Properties.Mask.EditMask = "N0"
            editor.Properties.Mask.UseMaskAsDisplayFormat = True

            editor.Parent = Me
            trackBar.Parent = Me

            ' Set type of POCO-ViewModel
            mvvmContext.ViewModelType = GetType(ViewModel)
            ' Data binding for the Progress property (via MVVMContext FluentAPI)
            Dim fluent = mvvmContext.OfType(Of ViewModel)()
            ' Binding two integer properties - 'Value' and 'Progress'. No conversion needed
            fluent.SetBinding(trackBar, Function(t) t.Value, Function(x) x.Progress)
            ' Binding the string 'Text' property to the integer 'Progress'. Values are automatically converted to/from an appropriate type.
            fluent.SetBinding(editor, Function(e) e.Text, Function(x) x.Progress)
        End Sub
        Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        Public Class ViewModel
            Public Overridable Property Progress() As Integer
        End Class
    End Class
End Namespace